src/clients/SpokePoolClient/SpokePoolClient.ts

```diff
@@ -86,6 +87,7 @@ export abstract class SpokePoolClient extends BaseAbstractClient {
   protected relayerRefundExecutions: RelayerRefundExecutionWithBlock[] = [];
   protected configStoreClient: AcrossConfigStoreClient | undefined;
   protected invalidFills: Set<string> = new Set();
+  protected spokePoolAddress: Address | undefined;
```

Comment:
Do we want this to be protected? We do have some use cases where we grab the address via `SpokePoolClient.spokePool.address`, but we can't rely on that working for SVM. If  `address` is public then we could read it directly.

Also, is it possible to be `undefined` because we only set it in the extended (i.e. EVM/SVM) constructor? That could be annoying for any users; maybe we could have an abstract method `getAddress()` that the `SpokePoolClient` calls, but which is implemented in the EVM/SVM implementations.

```suggestion
  readonly address: Address | undefined;
```

Reply:
Good catch. That's what this PR was meant to address https://github.com/across-protocol/sdk/pull/1052/commits/71d591f2bf8c678bdee0cd83e07610f339430b33